org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/rename/RenameService2.java

```diff
@@ -197,6 +198,20 @@ protected boolean isIdentifier(ILeafNode leafNode) {
 				|| leafNode.getGrammarElement() instanceof CrossReference)
 				&& !tokenUtil.isWhitespaceOrCommentNode(leafNode);
 	}
+	
+	/**
+	 * Override this method to supplement custom modifications on a {@link WorkspaceEdit}. 
+	 * This method is called after every other modification has been processed.
+	 * 
+	 * @param eObject Originally renamed EObject
+	 * @param edit Allows for additional changes to be added 
+	 * @param languageServerAccess Access to the language server
+	 * 
+	 * @since 2.25
+	 */
+	protected void applyCustomModifications(EObject eObject, WorkspaceEdit edit, ILanguageServerAccess languageServerAccess) {
```

Comment:
The rename options themselves would be handy.
It'd also make sense to move the usage of the ChangeSerializer (lines 128-141) into an own method that can be specialized for languages, that do not have other means to perform renames. This would probably render the applyCustomModifications obsolete since custom stuff could be done by overriding + calling super

Reply:
Thanks for the feedback. I'd agree, that's definitely a cleaner solution. I've implemented it this way.